### PR TITLE
Add ConVar.GetDescription() method

### PR DESCRIPTION
**Pull request by Wend4r**
Issue #1432

### CODE REVIEW

**Comment by Wend4r** [comment]
Works
image

File: plugins/include/convars.inc [hunk 1]
```diff
@@ -446,6 +452,16 @@ native void SetConVarBounds(Handle convar, ConVarBounds type, bool set, float va
...
 
+/**
+ * Retrieves the description of a console variable.
+ *
+ * @param convar        Handle to the convar.
+ * @param buffer        Buffer to store the description of the convar.
+ * @param maxlength     Maximum length of string buffer.
+ * @error               Invalid or corrupt Handle.
+ */
+native void GetConVarDescription(Handle convar, char[] buffer, int maxlength);
```
Comment (Headline) [inline plugins/include/convars.inc:463]: I understand adding this for the sake of completion, but we don't add old style functions for new features. Rest looks good to me, though! 😄

**Review by Headline — APPROVED** [review]

**Comment by peace-maker** [comment]
We need to figure out how to handle changes to translation phrases before merging this!

**Comment by Wend4r** [comment]
> We need to figure out how to handle changes to translation phrases before merging this!
I followed the standards set by `basecommands`
https://github.com/alliedmodders/sourcemod/blob/ad485069a837f602bdeeeb50f9e02452b3860ecd/plugins/basecommands.sp#L65-L66
Phrases from ConVars are found in these 2 files.
I think this already needs to be fixed with a separate solution.

**Comment by peace-maker** [comment]
Yes, sorry, this is nothing wrong with your changes. Just like #597 changes to the translation files have to be reflected in the [translator tool](https://sm.alliedmods.net/translator/) used to gather all the different languages. That update isn't automated yet and the [process of translation](https://wiki.alliedmods.net/Translating_SourceMod) hasn't been revisited since SourceMod switched to a rolling release cycle afaik.

**Comment by asherkin** [comment]
I think it would be best to remove the basecommands changes so we can get the new API merged.

File: core/smn_console.cpp [hunk 2]
```diff
@@ -650,6 +650,23 @@ static cell_t sm_GetConVarName(IPluginContext *pContext, const cell_t *params)
...
+	HandleError err;
+	ConVar *pConVar;
+
+	if ((err=g_ConVarManager.ReadConVarHandle(hndl, &pConVar))
+		!= HandleError_None)
+	{
+		return pContext->ThrowNativeError("Invalid convar handle %x (error %d)", hndl, err);
+	}
+
+	pContext->StringToLocalUTF8(params[2], params[3], pConVar->GetHelpText(), NULL);
```
Comment (peace-maker) [inline core/smn_console.cpp:665]: Can `GetHelpText` ever return NULL?

**Review by peace-maker — CHANGES_REQUESTED** [review]
The non-methodmap native needs to go, but otherwise this looks great!

File: core/smn_console.cpp (hunk 2 above)
Comment (Wend4r) [reply core/smn_console.cpp:665]: It will return a pointer with an empty string if there is no description

Comment (Wend4r) [reply core/smn_console.cpp:665]: https://github.com/alliedmodders/hl2sdk/blob/csgo/tier1/convar.cpp#L163

**Comment by Wend4r** [comment]
When there was an active change with `basecommands` it returns an empty string aka NULL_STRING from `ConVar.GetDescription()` from ConVars without a description.

File: core/smn_console.cpp (hunk 2 above)
Comment (peace-maker) [reply core/smn_console.cpp:665]: Cool, let's assume this is the case for all games and sdks

**Review by peace-maker — APPROVED** [review]

**Comment by peace-maker** [comment]
`NULL_STRING` and an empty string aren't the same, but I guess you're talking about an empty string, so this is fine.

**Pull request merged**